backtrace-library/src/androidTest/java/backtraceio/library/BacktraceClientMetricsTest.java

```diff
@@ -0,0 +1,938 @@
+        try {
+            sleep(100);
+        } catch (Exception e) {
+            fail(e.toString());
+        }
+        assertEquals(0, mockRequestHandler.numAttempts);
+    }
+
+    @Test
+    public void uploadSummedEventsManual() {
+        final Waiter waiter = new Waiter();
+
+        backtraceClient.metrics.enable(new BacktraceMetricsSettings(universeName, token, defaultBaseUrl, 0));
+        MockRequestHandler mockRequestHandler = new MockRequestHandler();
+        backtraceClient.metrics.setSummedEventsRequestHandler(mockRequestHandler);
+
+        backtraceClient.setSummedEventsOnServerResponse(new EventsOnServerResponseEventListener() {
+            @Override
+            public void onEvent(EventsResult result) {
+                assertEquals(0, result.getEventsPayload().getDroppedEvents());
+                String eventsPayloadJson = BacktraceSerializeHelper.toJson(result.getEventsPayload());
+                assertNotEquals(0, eventsPayloadJson.length());
+                assertEquals(BacktraceResultStatus.Ok, result.status);
+                waiter.resume();
+            }
+        });
+
+        backtraceClient.metrics.addSummedEvent(summedEventName);
+
+        assertEquals(1, backtraceClient.metrics.getSummedEvents().size());
+        // We will always have startup unique event GUID
+        assertEquals(1, backtraceClient.metrics.getUniqueEvents().size());
+
+        backtraceClient.metrics.send();
+
+        try {
+            waiter.await(1000);
+        } catch (Exception e) {
+            fail(e.toString());
+        }
+
+        assertNotEquals(0, mockRequestHandler.lastEventPayloadJson.length());
+        assertEquals(1, mockRequestHandler.numAttempts);
+        assertEquals(0, backtraceClient.metrics.getSummedEvents().size());
+        // We will always have startup unique event GUID
+        assertEquals(1, backtraceClient.metrics.getUniqueEvents().size());
+    }
+
+    @Test
+    public void uploadUniqueEventsManual() {
+        final Waiter waiter = new Waiter();
+
+        backtraceClient.metrics.enable(new BacktraceMetricsSettings(universeName, token, defaultBaseUrl, 0));
+        MockRequestHandler mockRequestHandler = new MockRequestHandler();
+        backtraceClient.metrics.setUniqueEventsRequestHandler(mockRequestHandler);
+
+        backtraceClient.setUniqueEventsOnServerResponse(new EventsOnServerResponseEventListener() {
+            @Override
+            public void onEvent(EventsResult result) {
+                assertEquals(0, result.getEventsPayload().getDroppedEvents());
+                String eventsPayloadJson = BacktraceSerializeHelper.toJson(result.getEventsPayload());
+                assertNotEquals(0, eventsPayloadJson.length());
+                assertEquals(BacktraceResultStatus.Ok, result.status);
+                waiter.resume();
+            }
+        });
+
+        backtraceClient.metrics.addUniqueEvent(uniqueAttributeName[0]);
```

Comment:
do we need at all to add unique event? By default we should have at least one default unique event.  Also - I think it's useful to check if unique event attribute is valid in the end of the test.

Reply:
Should I make a new test to check that once we add unique event it will send both events, default unique event and my custom unique event @konraddysput ?

Reply:
I think we can have separated unit test in my opinion.

Reply:
Done, thank you!